figures/serializers.py

```diff
@@ -749,15 +749,16 @@ class Meta:
 
     def get_registration_fields(self, user):
         registration_fields = dict()
-        user_profile = user.profile
-        for user_field in user_profile._meta.fields:
-            if user_field.name in self.context['required_fields']:
-                if user_field.choices:
-                    value = user_profile._get_FIELD_display(user_field)
-                else:
-                    value = getattr(user_profile, user_field.name)
-
-                registration_fields[user_field.name] = value
+        if getattr(user, 'profile', None):
+            user_profile = user.profile
+            for user_field in user_profile._meta.fields:
```

Comment:
how about we change it to this to reduce indentation:
```
user_profile = getattr(user, 'profile', None)
if not user_profile:
    return registration_fields
    
for user_field in user_profile._meta.fields:
```

Reply:
In either way we are doing the same thing, however, your solution will reduce code indentation.